test/jdk/java/awt/MenuItem/NullOrEmptyStringLabelTest.java

```diff
@@ -0,0 +1,102 @@
+ * questions.
+ */
+
+import java.awt.Button;
+import java.awt.Frame;
+import java.awt.Menu;
+import java.awt.MenuBar;
+import java.awt.MenuItem;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+/*
+ * @test
+ * @bug 4251036
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @summary MenuItem setLabel(null/"") behaves differently under Win32 and Solaris
+ * @run main/manual NullOrEmptyStringLabelTest
+ */
+
+public class NullOrEmptyStringLabelTest {
+
+    public static void main(String[] args) throws Exception {
+        String INSTRUCTIONS = """
+                The bug is reproducible under Win32 and Solaris.
+                Setting 'null' and "" as a label of menu item
+                should by the spec set blank label on all platforms.
+                But under Solaris setting "" as a label of menu item used to
+                cause some garbage to be set as label.
+                Under Win32 setting 'null' as a label used to result in
+                throwing NullPointerException.
+
+                If you see any of these things happen test fails otherwise
+                it passes.""";
+
+        PassFailJFrame.builder()
+                .title("NullOrEmptyStringLabelTest")
+                .instructions(INSTRUCTIONS)
+                .rows((int) INSTRUCTIONS.lines().count() + 2)
+                .columns(40)
+                .testTimeOut(5)
+                .testUI(NullOrEmptyStringLabelTest::createAndShowUI)
+                .build()
+                .awaitAndCheck();
+    }
+
+    private static Frame createAndShowUI() {
+        Frame frame = new Frame("Null Or Empty String Label Test");
+        Menu menu = new Menu("Menu");
+        MenuItem mi = new MenuItem("Item");
+        MenuBar mb = new MenuBar();
+        Button button1 = new Button("Set MenuItem label to 'null'");
+        Button button2 = new Button("Set MenuItem label to \"\"");
+        Button button3 = new Button("Set MenuItem label to 'text'");
+        button1.addActionListener(new ActionListener() {
+            public void actionPerformed(ActionEvent ev) {
+                System.out.println("Setting MenuItem label to null");
+                mi.setLabel(null);
+            }
+        });
+        button2.addActionListener(new ActionListener() {
+            public void actionPerformed(ActionEvent ev) {
+                System.out.println("Setting MenuItem label to \"\"");
+                mi.setLabel("");
+            }
+        });
+        button3.addActionListener(new ActionListener() {
+            public void actionPerformed(ActionEvent ev) {
+                System.out.println("Setting MenuItem label to 'text'");
+                mi.setLabel("text");
+            }
+        });
+        menu.add(mi);
+        mb.add(menu);
+        frame.add("North", button1);
+        frame.add("South", button2);
+        frame.add("Center", button3);
```

Comment:
Wouldn't it be better to keep the logical and visual order of the buttons consistent? Or is the intention to set the label to `"text"` after `null` followed by `""`?

I also recommend using newer syntax (order of parameters) and constants from `BorderLayout`:
```suggestion
        frame.add(button1, BorderLayout.NORTH);
        frame.add(button2, BorderLayout.CENTER);
        frame.add(button3, BorderLayout.SOUTH);
```

Reply:
>Wouldn't it be better to keep the logical and visual order of the buttons consistent? Or is the intention to set the label to "text" after null followed by ""?

I Just left them as it was before conversion.